go/private/context.bzl

```diff
@@ -995,6 +1004,11 @@ def _go_config_impl(ctx):
 
     toolchain = ctx.toolchains[GO_TOOLCHAIN]
 
+    # Mirror Go's logic by defaulting to PIE on supported platforms
+    linkmode = ctx.attr.linkmode[BuildSettingInfo].value
+    if linkmode == LINKMODE_NORMAL and _defaults_to_pie(toolchain.default_goos, race):
```

Comment:
Shouldn't we do this only if the `linkmode` flag is set to `auto`? Is there a way to force this back to `normal` after this PR?